### ISSUE TITLE
All operations are very slow [Home Builder 4.0.2]

### ISSUE THREAD

**Issue by Jsewill** [CreativeDesigner3D/home_builder_4#7]
I'm running blender 4.1 on arch linux. Every Home Builder operation in the 3D view, that I have tried, is extremely slow, especially dragging products. It only gets worse as more items are added to the scene. It looks like an awesome addon, and I'd like to get it functioning properly on my system.

I tried it on a windows machine, and it is not slow at all.

**Comment by Jsewill** [comment]
Now you have heard of it. There's also an issue on the Home Builder 3 repository that mentions it too: https://github.com/CreativeDesigner3D/home_builder/issues/4

The official build produces the same results. New products are almost unusable after a certain point.

**Comment by CreativeDesigner3D** [comment]
I'm not sure why it would work fine on windows and not Linux. I will do some research. I am not very familiar with Linux, so it might take me a bit to figure out what is going on.

**Comment by Christian-Prather** [comment]
I am also experiencing this with Blender 4.1.1 and home builder. 

Walls and dragging in all products lags, is unresponsive and many times crashes Blender. 
Running Ubuntu 22.04

Happy to help fix / give more info if there is a place you suspect might be the issue. Never done a blender add on but experienced with Linux and a software dev if you would like some assistance as I love the tool.

**Comment by CreativeDesigner3D** [comment]
Yeah I would love some help determining what the issue is. Everything is working fine on Windows. I would expect the issue has something to do with the ray cast function in the Blender Python API. I don't know why that would work differently on Linux.

**Comment by Christian-Prather** [comment]
Ya I am running it on windows currently and it is very snappy. I havent looked too deep into your code is there a particular section you suspect that calls into that API? I can dive more into it this afternoon and see if I notice anything obvious.

**Comment by CreativeDesigner3D** [comment]
The module python_libs\pc_lib\pc_snap.py is the main file that calculates the snap location during placement. If the issue is with ray cast that would be a good place to start. Thanks.

**Comment by Christian-Prather** [comment]
Okay sounds good ill dive in and see what I learn.

**Comment by Anaminus** [comment]
This seems to be caused by the addon responding to an excessive number of events. While using the wall tool, I get a lot of INBETWEEN_MOUSEMOVE events emitted between each regular MOUSEMOVE, leading to a lot of unnecessary updates. Maybe these events are produced on Linux but not Windows, which is why there's a difference.

Ignoring INBETWEEN_MOUSEMOVE events appears to eliminate the problem. For example:

```python
...
    def modal(self, context, event):
        if event.type == "INBETWEEN_MOUSEMOVE":
            return {'RUNNING_MODAL'}
...
```

**Comment by jmtatsch** [comment]
Same issue on mac. The more walls, windows and doors it gets painfully slow quickly when dragging out  windows etc. And my project is not really complex.

**Comment by chaxor** [comment]
This is a major problem that makes the software effectively not usable. @Anaminus where do you put this very simple fix? Can you make a pull request so this can be fixed for the huge % of users this affects? I don't really know many people that still use windows.
